### ISSUE TITLE
Figure: Reproduction of paper data

### ISSUE THREAD

**Issue by kaylai** [kaylai/VESIcal#39]
Make a figure showing how reported data in the literature compares to that calculated using our code, to prove that it's working properly.

**Comment by kaylai** [comment]
Made these for MooreWater() in a separate notebook. Would be nice to have figures for every model for every core calculation that we could put in an appendix.

Penny, is this something you could do alongside the calibration range figures, since I guess it'll use the same data, right?

**Comment by PennyWieser** [comment]
@kaylai to clarify, do you want me to find studies that have used these solubility models on natural data (e.g., the bennett paper) or do you want me to run each solubility model for its own calibration dataset and see how the P of the experiment compares to the P we calculate using that model? Or both?

**Comment by simonwmatthews** [comment]
I guess the ideal thing (I think?) would be to compare calculation results between VESIcal and results shown in the model papers. Though a lot of these will just be lines on plots, which would be slightly awkward to extract. This avoids having to worry about which version of a code another study has implemented etc (and what assumptions they've made when running it, if not reported). In an ideal world (and choosing the right datasets?) this won't be a problem...

I am also aware of at least one model paper where the model results are inconsistent between figures within that paper... I wasted a day trying to reproduce those figures...

**Comment by kaylai** [comment]
@PennyWieser I think the latter... so, comparing our results on a model calibration set to reported values for that set. The results should fall around a 1:1 line.

I've done this for MooreWater() here: https://github.com/kaylai/VESIcal/blob/master/testDataSets/MooreWaterTest.ipynb

Although you'll see there are issues with the equilibrium fluid comp plot (I think I know why... so please ignore for now), but you can see how the other two plots look great. We could do this kind of exercise for each model.

**Comment by PennyWieser** [comment]
@simonwmatthews will you let me know when you fix the Celcius/Kelvin issue so that I can continue with my checks of different models vs. paper data. 

@kaylai for Magmasat, its really hard to get the calibration dataset, Mark Ghiorso described it as as "stuck in purgatory" during a move of the LEPR server - For the TAS diagram, I manually extracted the points from their diagram, as from the base publications they cite, it really wasn't clear what experiments were and weren't included (it just says how many, not which ones). Do you think I should just instead make a figure for the supplement using Emma Bennett's Nature dataset where she used the Mac App to show the results match? 

Same problem with the Dixon calibration dataset - a lot is unpublished data, particularly for H2O, and Dixon can't get at it at the moment. There is also no mixed H2O-CO2 stuff in the ones i've compiled. Given we are just recreating VolatileCalc, should I just plug some test compositions into that? The only thing we would need to change is the current way the code is keeping SiO2 at 48.9, while volatilecalc does it at 49. 

Thoughts/suggestions welcome on best way to do these ones!

**Comment by simonwmatthews** [comment]
@PennyWieser I will try to get to this as soon as possible, but I'm still really pushed for time this week, sorry. The line "temperature = temperature + 273.15" needs to be added at the start of each method, and the docstrings updated to specify the temperature is in degC not K. And the calibration_range objects will need changing to degC for each method.

**Comment by kaylai** [comment]
> @PennyWieser I will try to get to this as soon as possible, but I'm still really pushed for time this week, sorry. The line "temperature = temperature + 273.15" needs to be added at the start of each method, and the docstrings updated to specify the temperature is in degC not K. And the calibration_range objects will need changing to degC for each model.

This is now complete.

**Comment by kaylai** [comment]
@PennyWieser 

>Do you think I should just instead make a figure for the supplement using Emma Bennett's Nature dataset where she used the Mac App to show the results match?
> 
> Same problem with the Dixon calibration dataset.. Given we are just recreating VolatileCalc, should I just plug some test compositions into that? The only thing we would need to change is the current way the code is keeping SiO2 at 48.9, while volatilecalc does it at 49.

Yes, I think these are both very good solutions! Just something to show that the results we get from our model match the results we get from the original models. Thanks!

**Comment by PennyWieser** [comment]
Great, final few questions then i'll finish these tomorow - thanks for sorting the Temp thing!

 1. Do we want these comparisons in separate notebooks as you did for the Moore one (e.g. one for each model), and then just bring the figures into the manuscript as pngs or whatever to avoid clutter? Similarly, for the calibration P, T, X range of compositions table, I see at the moment you're bringing it in as a spreadsheet, do we want to keep it like this as just a dataframe, or should I make a prettier png/vector image file of a table (e.g., maybe rows color coded for the different models as in the calibration TAS diagrams etc). 

Final point, do we want to unify our plotting environments to look the same throughout the manuscript? e.g., font size, line width, font style, ticks etc? If we agree on one, I can make sure I follow it in all the separate workbooks, then we can just add something defining it in the manuscript so all the figures made throughout match the figures pulled in as pngs. E.g., the one i'm currently using is
![image](https://user-images.githubusercontent.com/65668673/84813700-3db94380-b008-11ea-9fd5-4c028fe8471b.png)

**Comment by kaylai** [comment]
Penny,

Awesome, sounds good. 

For the comparisons:
Perhaps the best thing to do for now is to put the comparisons in separate notebooks as you say, but also save the plots as images that could be put either in the main manuscript or in a supplemental file. We can then reference the notebooks so that folks can see how the plots were generated. 

For the tables:
Yes, perhaps what you suggest makes the most sense. We can provide the xlsx file as a supplemental dataset, but then we can use a pretty image of a table as the table in the manuscript itself (referencing the supp dataset in the table caption)

For plot style:
Yes I think that’s a great idea. If you have some thoughts please feel free to make a style. I can also give that some thought. Would be cool to have a consistent VESIcal color scheme. I LOVE the idea of someone seeing a plot and going “oh I can tell that’s from VESIcal due to the style.” One place to start might be the colors used in the two figures I’ve made thus far (the schematic diagrams), which use the same color palette.

**Comment by PennyWieser** [comment]
@kaylai @simonwmatthews is there an easy way to adapt class ExcelFile(object) to allow you to specify the sheet name at this stage?
 I've had a few goes but no success... I was planning on reading in the "solubility dataset" file as is and putting it through the different models, rather than having separately formatted spreadsheets for each comparison to save the total number of files people need to download.

**Comment by kaylai** [comment]
@PennyWieser Implemented! 

Use example:

```python
myfile = v.ExcelFile("path/to/your/data.xlsx")
dissolved = myfile.calculate_dissolved_volatiles(args)
someothercalc = myfile.calculate_equilibrium_fluid_comp(args)
myfile.save_excelfile("mynewfilename.xlsx", calculations=[dissolved, someothercalc], sheet_name=["A dissolved sheet", "Some other sheet"])
```

Commit 074d8ee0485d04f816489e8340f89f7e0b4c94c1

**Comment by PennyWieser** [comment]
Calling mac users @simonwmatthews @kaylai 

There are a few points in the Bennet dataset that show slightly larger offsets from VESIcal. 
![image](https://user-images.githubusercontent.com/65668673/84930023-994b0600-b0c8-11ea-862b-22e68cb25b3e.png)
The max is 55 bars which is still pretty small. I'm trying to think the best way to explain these:
1) slight differences in inputs -e.g., perhaps emma entered a constant MnO content, while i'm putting in 0 as she doesnt give any data for any inclusions. However, as the difference is symmetrical, I don't think this is likely as it would be systematic.
2) she typed a few of them wrong. But again, a lot are scattered. Perhaps the 55 bar one is a typo?

Basically, are we worried about offsets of this magnitude? I tried entering the composition with the biggest offset into the web calculator, but it won't converge because that has stopped working on PCs. If it is of concern, if either of you have the mac app could you possibly check the PSat for the yellow composition? I can send you it as a spreadsheet to copy-paste if thats easier

![image](https://user-images.githubusercontent.com/65668673/84930085-af58c680-b0c8-11ea-8ea4-067b35c0b082.png)

**Comment by kaylai** [comment]
> Sorry, that wasn't clear, I meant loading them, not saving them? At the moment if I do
>
>myfile = v.ExcelFile('Solubility_Datasets_V1.xlsx', input_type='wtpercent')
>
>I can't get it to do anything other than read the first sheet, even if I add a sheet_name="" here and at various points in VESIcal (although almost certainly the wrong places!)

Ahhh yes of course. This is a good point. Will update this to give an option to select a worksheet by name, else it will default to the first worksheet.

**Comment by kaylai** [comment]
Implemented!

Doc string:
>sheet_name: str
			OPTIONAL. Default value is 0 which gets the first sheet in the excel spreadsheet file. This implements the pandas.
			read_excel() sheet_name parameter. But functionality to read in more than one sheet at a time (e.g., pandas.read_excel(sheet_name=None))
			is not yet imlpemented in VESIcal. From the pandas 1.0.4 documentation:
				Available cases:
					- Defaults to 0: 1st sheet as a DataFrame
					- 1: 2nd sheet as a DataFrame
					- "Sheet1": Load sheet with name “Sheet1”

Usage:

```python
import VESIcal as v
myfile_sheet1 = v.ExcelFile("test_ex.xlsx", sheet_name=0)
myfile_sheet2 = v.ExcelFile("test_ex.xslx", sheet_name=1)
```

OR you can pass the sheet name as a string:
```python
import VESIcal as v
myfile_sheet1 = v.ExcelFile("test_ex.xlsx", sheet_name="NameOfTheFirstSheet")
myfile_sheet2 = v.ExcelFile("test_ex.xslx", sheet_name="NameOfTheSecondSheet")
```

**Comment by kaylai** [comment]
> it is of concern, if either of you have the mac app could you possibly check the PSat for the yellow composition? I can send you it as a spreadsheet to copy-paste if thats easier

This looks great! I'm not too bothered by the one point. This is comparing VESIcal to MagmaSat, right? If so I can easily run the other point on my mac.

**Comment by PennyWieser** [comment]
@kaylai thanks for rerunning it, Emma must have typed a number in wrong/copied PSat wrong over, that matches now. I'm amazed she only made 1 mistake manually entering this much! A very commendable effort.

**Comment by PennyWieser** [comment]
@kaylai and @simonwmatthews before I close this issue, what is the plan with all these notebooks I've made  now we are so close to getting everything matching- just supply one notebook + excel file per model so people can run them themselves or just loook through the results? Or also export all the figs into a document and run through the tests I did for each model and justify it all works? Given that jupyter saves the graphs, I'm tempted to say we could, at worst, export 1 pdf doc per model as the test, but ideally just keep them as notebooks.
